tenacity/_asyncio.py

```diff
@@ -90,3 +93,48 @@ async def async_wrapped(*args: typing.Any, **kwargs: typing.Any) -> typing.Any:
         async_wrapped.retry_with = fn.retry_with
 
         return async_wrapped
+
+    @staticmethod
+    async def handle_custom_function(
+        func: typing.Union[typing.Callable, typing.Awaitable], retry_state: RetryCallState
+    ) -> typing.Any:
+        if iscoroutinefunction(func):
+            return await func(retry_state)
+        return func(retry_state)
+
+    async def iter(self, retry_state: "RetryCallState") -> typing.Union[DoAttempt, DoSleep, typing.Any]:  # noqa
```

Comment:
@jd I also wanted to use some base method to avoid re-defining the iter method again, however, one of them is sync `def iter` and other one is async `async def iter`. Any ideas how to make it DRY?

I wanted to use `handle_custom_function` to handle sync/async calls, however, if we make `handle_custom_function` sync function, then it might need to use `asyncio.get_event_loop()` to run awaitable functions (running async functions from sync function).

Reply:
@jd any ideas? I would love to have progress with this PR and start using async callbacks for `retry_error_callback`